src/native/managed/compile-native.proj

```diff
@@ -0,0 +1,67 @@
+        <SupportsNativeAotComponents Condition="'$(SupportsNativeAotComponents)' == ''">false</SupportsNativeAotComponents>
+    </PropertyGroup>
+
+    <!-- some special kinds of runtime builds need extra NativeAOT flags -->
+    <PropertyGroup>
+        <SysRoot Condition="'$(CrossBuild)' == 'true' and '$(HostOS)' != 'windows'">$(ROOTFS_DIR)</SysRoot>
+        <LinkerFlavor Condition="'$(CrossBuild)' == 'true' and '$(TargetsLinux)' == 'true'">lld</LinkerFlavor>
+        <CustomLinkerArgToolchainArg Condition="'$(CrossBuild)' == 'true' and '$(_hostArchitecture)' == '$(_targetArchitecture)' and '$(_hostOS)' != 'windows'">--gcc-toolchain=$(ROOTFS_DIR)/usr</CustomLinkerArgToolchainArg>
+    </PropertyGroup>
+
+    <!-- properties to pass down to the subproject builds -->
+    <ItemGroup>
+        <SubprojectProps Include="Configuration" Value="$(NativeLibsPublishConfiguration)" />
+        <SubprojectProps Include="RuntimeConfiguration" Value="$(RuntimeConfiguration)" />
+        <SubprojectProps Include="LibrariesConfiguration" Value="$(LibrariesConfiguration)" />
+        <SubprojectProps Include="RuntimeIdentifier" Value="$(OutputRID)" />
+
+        <SubprojectProps Include="NativeLib" Value="$(NativeLibKind)" />
+
+        <SubprojectProps Condition="'$(SysRoot)' != ''" Include="SysRoot" Value="$(SysRoot)" />
+        <SubprojectProps Condition="'$(LinkerFlavor)' != ''" Include="LinkerFlavor" Value="$(LinkerFlavor)" />
+        <SubprojectProps Condition="'$(CustomLinkerArgToolchainArg)' != ''" Include="CustomLinkerArgToolchainArg" Value="$(CustomLinkerArgToolchainArg)" />
+    </ItemGroup>
+
+    <ItemGroup Condition="$(SupportsNativeAotComponents)">
+        <ProjectReference Include="@(NativeLibsProjectsToBuild)"
+                          ReferenceOutputAssembly="false"
+                          AdditionalProperties="%(AdditionalProperties);@(SubprojectProps->'%(Identity)=%(Value)', ';')"/>
+    </ItemGroup>
+
+    <Target Name="BuildNativeLibs" BeforeTargets="Build" Condition="$(SupportsNativeAotComponents)">
+        <PropertyGroup>
+            <SplitSubprojectProps>@(SubprojectProps->'%(Identity)=%(Value)', ';')</SplitSubprojectProps>
+        </PropertyGroup>
+
+        <!-- workaround https://github.com/dotnet/msbuild/issues/2811 - Restore has to be a separate exectution with a different evaluation -->
+        <MSBuild Projects="@(NativeLibsProjectsToBuild)"
+                 Properties="MSBuildRestoreSessionId=$([System.Guid]::NewGuid());$(SplitSubprojectProps)"
+                 Targets="Restore" />
+
+        <MSBuild Projects="@(NativeLibsProjectsToBuild)"
+                 Properties="_IsPublishing=true;$(SplitSubprojectProps)"
+                 Targets="Build;Publish" />
+    </Target>
```

Comment:
We should be able to reference the project just through a `ProjectReference`. Additionally, by making Publish the default target above, then we should automatically publish here when `runtime-prereqs.proj` references this through a ProjectReference (the build step would call the `Publish` target)

```suggestion
    <ItemGroup>
	      <ProjectReference Include="@(NativeLibsProjectsToBuild)" AdditionalProperties="$(SplitSubprojectProps)" Condition="$(SupportsNativeAotComponents)" />
    </ItemGroup>
```